sqlmesh/core/config/naming.py

```diff
@@ -0,0 +1,14 @@
+from __future__ import annotations
+
+from sqlmesh.core.config.base import BaseConfig
+
+
+class NameInferenceConfig(BaseConfig):
+    """Configuration for name inference of models from directory structure.
+
+    Args:
+        name_inference: A flag indicating whether name inference is enabled.
+                        Default is False.
+    """
+
+    name_inference: bool = False
```

Comment:
nit: can we replace this and all refs to infer_names

Reply:
Sure done!